src/engine/controls/keycontrol.cpp

```diff
@@ -476,3 +525,106 @@ bool KeyControl::syncKey(EngineBuffer* pOtherEngineBuffer) {
     slotPitchChanged(pitchToTakeOctaves * 12);
     return true;
 }
+
+void KeyControl::slot432HzPitchLockChanged(double value) {
+    Q_UNUSED(value);
+    apply432HzPitchAdjustment();
+}
+
+void KeyControl::slotFileIs432HzChanged(double value) {
+    Q_UNUSED(value);
+    apply432HzPitchAdjustment();
+}
+
+void KeyControl::slotFileTuningFrequencyChanged(double value) {
+    Q_UNUSED(value);
+    apply432HzPitchAdjustment();
+}
+
+void KeyControl::ensureFileTuningControls() {
+    if (!m_pFileIs432Hz &&
```

Comment:
Creating controls is expensive and a looking operation. That's why we crate all of them in the constructor, even if they arr not needed. Showing them conditionally is a Skin/Mapping issue.